### PR TITLE
docs: replace absolute hyperlinks to relative

### LINKED ISSUE #2404: Minor improvement for the website: use relative paths instead of absolute on hyperlinks to another page
There are several pages with hyperlinks to another page of the docs like this:

https://github.com/nestjs/docs.nestjs.com/blob/d15ab28fe673d7d3cc4e6ebfb3cbf053556f7f90/content/techniques/caching.md?plain=1#L97

but using `https://docs.nestjs.com/` instead of `../../` isn't good for those looking at the docs of another version (like v8) because they'll be redirect to the docs of the active version instead:

https://user-images.githubusercontent.com/13461315/179424127-1f16c63f-eaf0-4c5a-8357-ccd81e71234a.mp4

---

[Here is all ocurrences](https://cs.github.com/nestjs/docs.nestjs.com?q=%22https%3A%2F%2Fdocs.nestjs.com%2F%22+path%3Acontent) of `https://docs.nestjs.com/` under `content` directory:

image

changing them will be easy but there are a lot of files to manually touch.

I'm working on this issue already (slowly).

### PR DESCRIPTION
**Pull request by micalevisk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: closes #2404

## What is the new behavior?

Note that the preview resolves to `https://docs.nestjs.com/`

image

I'm not 100% sure if this PR fixes that issue because I didn't manage to switch to the v8 page locally

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by micalevisk** [comment]
I thought that using relative paths would be enough but I was wrong 

image

I need to investigate this more.

File: content/controllers.md [hunk 1]
```diff
@@ -256,10 +256,10 @@ Returned values will override any arguments passed to the `@Redirect()` decorato
...
 @Get('docs')
-@Redirect('https://docs.nestjs.com', 302)
+@Redirect('', 302)
 getDocs(@Query('version') version) {
   if (version && version === '5') {
-    return { url: 'https://docs.nestjs.com/v5/' };
+    return { url: '/v5/' };
   }
 }
 ```
```
Comment (Tony133) [inline content/controllers.md:265]: Does this code snippet also need to be changed?

Comment (micalevisk) [reply content/controllers.md:265]: nop, my bad. Thanks

Comment (Tony133) [reply content/controllers.md]: you're welcome, it's nothing 👍

**Review by Tony133 — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
AFAIR the idea here was that absolute links will open in a new window while relative in the same one

**Pull request closed without merging**